calico_cni/policy_drivers.py

```diff
@@ -338,28 +356,40 @@ def get_policy_driver(k8s_pod_name, k8s_namespace, net_config):
     policy_config = net_config.get(POLICY_KEY, {})
     network_name = net_config["name"]
     policy_type = policy_config.get("type")
+    supported_policy_types = [None,
+                              POLICY_MODE_KUBERNETES,
+                              POLICY_MODE_KUBERNETES_ANNOTATIONS]
+    if policy_type not in supported_policy_types:
+        print_cni_error(ERR_CODE_GENERIC,
+                        "policy type set to unsupported value (%s). "
+                        "Supported values = %s" %
+                        (policy_type, supported_policy_types))
+        sys.exit(ERR_CODE_GENERIC)
 
     # Determine which policy driver to use.
-    if policy_type == POLICY_MODE_DENY_INBOUND:
-        # Use the deny-inbound driver. 
-        driver_cls = DefaultDenyInboundDriver 
-        driver_args = [network_name]
-    elif k8s_pod_name:
-        # Runing under Kubernetes - decide which Kubernetes driver to use.
-        if policy_type == POLICY_MODE_ANNOTATIONS: 
-            _log.debug("Using Kubernetes Annotation Policy Driver")
+    if k8s_pod_name:
+        # Running under Kubernetes - decide which Kubernetes driver to use.
+        if policy_type == POLICY_MODE_KUBERNETES_ANNOTATIONS or \
+           policy_type == POLICY_MODE_KUBERNETES:
             assert k8s_namespace, "Missing Kubernetes namespace"
             k8s_auth_token = policy_config.get(AUTH_TOKEN_KEY)
-            k8s_api_root = policy_config.get(API_ROOT_KEY, 
+            k8s_api_root = policy_config.get(API_ROOT_KEY,
                                              "https://10.100.0.1:443/api/v1/")
-            driver_cls = KubernetesAnnotationDriver
-            driver_args = [k8s_pod_name, 
-                           k8s_namespace, 
+
+            if policy_type == POLICY_MODE_KUBERNETES:
+                _log.debug("Using Kubernetes Policy Driver")
+                driver_cls = KubernetesPolicyDriver
+            elif policy_type == POLICY_MODE_KUBERNETES_ANNOTATIONS:
+                _log.debug("Using Kubernetes Annotation Policy Driver")
+                driver_cls = KubernetesAnnotationDriver
+
+            driver_args = [k8s_pod_name,
+                           k8s_namespace,
                            k8s_auth_token,
                            k8s_api_root]
         else:
             _log.debug("Using Default Kubernetes Policy Driver")
```

Comment:
I might re-word this log now to say "Using Kubernetes Driver - no policy" or something like that to be more clear.